e2e_test/e2e_test.go

```diff
@@ -341,6 +341,71 @@ func TestForexPluginsHappyCase(t *testing.T) {
 	testHappyCase(t, o, endRound, pricePrecision)
 }
 
+func TestCAXPluginsHappyCase(t *testing.T) {
+	// run the test after the data source cax.devnet.clearmatics.network provides available data.
+	t.Skip("this test depends on the remote service endpoint of cax.devnet.clearmatics.network")
+	var conf = &NetworkConfig{
+		EnableL1Logs: false,
+		Symbols:      "NTN-USD,ATN-USD,NTN-ATN",
+		VotePeriod:   defaultVotePeriod,
+		PluginDIRs:   []string{caxPlugDir, caxPlugDir, caxPlugDir, caxPlugDir},
+	}
+
+	net, err := createNetwork(conf)
+	require.NoError(t, err)
+	defer net.Stop()
+
+	client, err := ethclient.Dial(fmt.Sprintf("ws://%s:%d", net.L1Nodes[0].Host, net.L1Nodes[0].WSPort))
+	require.NoError(t, err)
+	defer client.Close()
+
+	// bind client with oracle contract address
+	o, err := contract.NewOracle(types.OracleContractAddress, client)
+	require.NoError(t, err)
+
+	// subscribe on-chain round rotation event
+	chRoundEvent := make(chan *contract.OracleNewRound)
+	subRoundEvent, err := o.WatchNewRound(new(bind.WatchOpts), chRoundEvent)
+	require.NoError(t, err)
+	defer subRoundEvent.Unsubscribe()
+
+	symbols := []string{"NTN-USD", "ATN-USD", "NTN-ATN"}
+	endRound := uint64(5)
+
+	var prices []contract.IOracleRoundData
+
+	for {
+		select {
+		case rEvent := <-chRoundEvent:
+			if rEvent.Round.Uint64() <= 3 {
```

Comment:
Just to understand, why we wait for the first 3 rounds without doing anything?

Reply:
> Just to understand, why we wait for the first 3 rounds without doing anything?

Because in the e2e testing context, we want to have such 3-rounds period to get data reports from all clients be on-chain.